### PR TITLE
Add Universal Header component

### PR DESCRIPTION
**Pull request by jonearley**
## Todo
- [x] Setup component.
- [x] Fetch data and handle successful fetch.
- [x] Add responsive design.
- [x] Close when click outside dropdown.
- [x] Add loading indicator when fetching data.
- [x] Add failed to fetch data state.
- [x] Check in Safari, Chrome, Firefox, Edge, and IE 11.

### CODE REVIEW

File: packages/components/src/components/universal-header/universal-header.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+  }
+
+  @Listen("click", { target: "window" })
+  handleMouseDown(e) {
+    // Close on click outside of open dropdown, except the explore button.
+    if (this.dropdown) {
+      // This only works when not using the shadow dom.
+      // But how could we make this work with shadow dom enabled?
+      if (!this.dropdown.contains(e.target)) {
```
Comment (jonearley) [inline packages/components/src/components/universal-header/universal-header.tsx]: @respinos, @bridgetburke -- This doesn't work with shadow dom configured (`shadow: true`), but I'm not sure how else to close on click outside yet then... 🤔

Comment (respinos) [reply packages/components/src/components/universal-header/universal-header.tsx]: Is `this.dropdown` the host element? If I'm reading https://javascript.info/shadow-dom-events correctly, can you check `this.shadowRoow.contains`?

Comment (jonearley) [reply packages/components/src/components/universal-header/universal-header.tsx]: That makes sense! I'll try that. Oh, `this.dropdown` is not the host element, but a `ref` to the dropdown container.

Comment (respinos) [reply packages/components/src/components/universal-header/universal-header.tsx]: (Looks more closely at the code.) Huh, why wouldn't that work though? `this.dropdown` should be contained in `this.shadowRoot`.

Comment (jonearley) [reply packages/components/src/components/universal-header/universal-header.tsx]: Using `event.target.shadowRoot` I am not sure yet how to know if the click was inside the dropdown. I can check if the click was inside the universal header or not with `e.target.shadowRoot.contains(this.dropdown))`

Comment (jonearley) [reply packages/components/src/components/universal-header/universal-header.tsx]: (reading up on this a bit more still)

Comment (jonearley) [reply packages/components/src/components/universal-header/universal-header.tsx]: Ah-ha! `this.dropdown.contains(e.composedPath()[0])` this does the trick.

**Pull request merged**